use-cases/virgo/trainer.py

```diff
@@ -373,7 +370,279 @@ def train(self):
 
             # Report training metrics of last epoch to Ray
             train.report({"loss": np.mean(val_loss), "train_loss": np.mean(epoch_loss)})
+
+        return loss_plot, val_loss_plot, acc_plot, val_acc_plot
+
+
+class RayNoiseGeneratorTrainer(RayTorchTrainer):
+    def __init__(
+        self,
+        config: Dict,
+        strategy: Optional[Literal["ddp", "deepspeed", "horovod"]] = "ddp",
+        name: Optional[str] = None,
+        logger: Optional[Logger] = None,
+    ) -> None:
+        super().__init__(config=config, strategy=strategy, name=name, logger=logger)
+
+    def create_model_loss_optimizer(self) -> None:
+        # Select generator
+        generator = self.training_config["generator"]
+        if generator == "simple":
+            self.model = Decoder(3, norm=False)
+            init_weights(self.model, "normal", scaling=0.02)
+        elif generator == "deep":
+            self.model = Decoder_2d_deep(3)
+            init_weights(self.model, "normal", scaling=0.02)
+        elif generator == "resnet":
+            self.model = GeneratorResNet(3, 12, 1)
+            init_weights(self.model, "normal", scaling=0.01)
+        elif generator == "unet":
+            self.model = UNet(input_channels=3, output_channels=1, norm=False)
+            init_weights(self.model, "normal", scaling=0.02)
+        else:
+            raise ValueError("Unrecognized generator type! Got", generator)
+
+        # Select loss
+        loss = self.training_config["loss"]
+        if loss == "L1":
+            self.loss = nn.L1Loss()
+        elif loss == "L2":
+            self.loss = nn.MSELoss()
```

Comment:
Double check that capital letters are used for the loss everywhere, i.e. "L1" instead of "l1". I remember there being something in EURAC or something where these things were different.

Reply:
Yes this was always inconsistent in Virgo, I have changed everything to lowercase